lib/bamboo/adapters/smtp_adapter.ex

```diff
@@ -100,10 +100,18 @@ defmodule Bamboo.SMTPAdapter do
     {:ok, response}
   end
 
+  defp add_bcc(body, %Bamboo.Email{bcc: recipients}) when length(recipients) == 0 do
```

Comment:
I do not really know well this part of the code but could we guard with 
`defp add_bcc(body, %Bamboo.Email{bcc: []} do` instead of the `length/1` call ? 

Credo is not really happy with calling `length/1`

Reply:
Sure. I debated the best way to do that. I'll make the change and verify the tests.

Reply:
I did not see the `format_email_as_string` call that turns the list into a string in the other guard. Sorry for the false comment above.

Reply:
Actually, it calls https://github.com/fewlinesco/bamboo_smtp/blob/91d626021ac4789f044fea53c53c103fa6aa269d/lib/bamboo/adapters/smtp_adapter.ex#L173 since the cc and bcc are lists that are run through format_email_with_string (so they become a string from the list) when `format_email_as_string/2` is called. https://github.com/fewlinesco/bamboo_smtp/blob/91d626021ac4789f044fea53c53c103fa6aa269d/lib/bamboo/adapters/smtp_adapter.ex#L108. When the blanks lists are converted to string, the result is an empty string. So the headers appear to be blank values, since the string is empty..

Reply:
yep I just saw that 👍 again sorry for the misleading comment.

Reply:
No problem :)